src/gfn/gym/discrete_ebm.py

```diff
@@ -108,14 +108,12 @@ def update_masks(self, states: DiscreteStates) -> None:
         states.backward_masks[..., : self.ndim] = states.tensor == 0
         states.backward_masks[..., self.ndim : 2 * self.ndim] = states.tensor == 1
 
-    def make_random_states_tensor(self, batch_shape: Tuple) -> torch.Tensor:
+    def make_random_states_tensor(
+        self, batch_shape: Tuple, device: torch.device
```

Comment:
can we draw this information from `self.device` (i.e., whatever device the env is initialized with) by default?